toolchain/lower/file_context.cpp

```diff
@@ -390,6 +423,24 @@ auto FileContext::BuildFunctionBody(SemIR::FunctionId function_id,
                                     SemIR::SpecificId specific_id) -> void {
   const auto& body_block_ids = function.body_block_ids;
   CARBON_DCHECK(llvm_function, "LLVM Function not found when lowering body.");
+
+  if (function.cpp_decl) {
+    clang::FunctionDecl* cpp_def = function.cpp_decl->getDefinition();
+    CARBON_DCHECK(cpp_def, "No Clang function body found during lowering");
+
+    // Create the LLVM function (`CodeGenModule::GetOrCreateLLVMFunction()`) so
+    // that code generation (`CodeGenModule::EmitGlobal()`) would see this
+    // function name (`CodeGenModule::getMangledName()`), and will generate its
+    // definition.
+    CARBON_DCHECK(
+        cpp_code_generator_->GetAddrOfGlobal(clang::GlobalDecl(cpp_def),
+                                             /*isForDefinition=*/false));
+
+    // Emit the function code.
+    cpp_code_generator_->HandleTopLevelDecl(clang::DeclGroupRef(cpp_def));
```

Comment:
If this call is necessary, are we able to handle the case where an inline function calls another inline function? Eg:
```c++
// --- inline_function_with_call.h

inline void bar() {}
inline void foo() { bar(); }
```
```carbon
// --- import_inline_function_decl.carbon

library "[[@TEST_NAME]]";

import Cpp library "inline_function_decl.h";

fn MyF() {
  // Should cause both `foo` and `bar` to be emitted.
  Cpp.foo();
}
```